### PR TITLE
Merge Metaclient feature branch

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

#2237 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This PR merges all meta client changes to main branch after certification is finished. 

### Tests

- [X] The following tests are written for this issue:

NA. Please check each individual PR for new unit tests. 

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Comment by desaikomal** [comment]
Xiaoyuan, there are still pending changes in the branch. I know you have completed the certification testing but wanted to know what happens to all the new / remaining changes?

**Comment by xyuanlu** [comment]
All CRUD operation PRs are merged in feature branch. Let's start the effort to merge feature branch to master.

**Comment by desaikomal** [comment]
Congratulations !!! Do we have to create any specific documentation around MetaClient yet?

File: .github/workflows/Helix-PR-CI.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Helix PR CI
 on:
   pull_request:
-    branches: [ master ]
+    branches: [ master, metaclient ] # TODO: remove side branch
```
Comment (desaikomal) [inline .github/workflows/Helix-PR-CI.yml:4]: should you remove this??

**Review by desaikomal — APPROVED** [review]

File: .github/workflows/Helix-PR-CI.yml (hunk 1 above)
Comment (xyuanlu) [reply .github/workflows/Helix-PR-CI.yml:4]: Will remove in future commit. Will keep the merge PR pure for merge.

**Pull request merged**